plugins/smartvol/transactions.go

```diff
@@ -0,0 +1,167 @@
+package smartvol
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	smartvolapi "github.com/gluster/glusterd2/plugins/smartvol/api"
+
+	"github.com/pborman/uuid"
+)
+
+func txnPrepareBricks(c transaction.TxnCtx) error {
+	var req smartvolapi.VolCreateReq
+	if err := c.Get("reqBricksPrepare", &req); err != nil {
+		c.Logger().WithError(err).WithField(
+			"key", "reqBricksPrepare").Debug("Failed to get key from store")
+		return err
+	}
+
+	if req.Size == 0 {
+		return nil
+	}
+
+	for _, sv := range req.Subvols {
+		for _, b := range sv.Bricks {
+			peerid := uuid.Parse(b.PeerID)
+			if peerid == nil {
+				// TODO: Peer ID Parse error
+			}
+
+			if !uuid.Equal(peerid, gdctx.MyUUID) {
+				continue
+			}
+
+			brickMountDir := path.Dir(b.Path)
+
+			// Create Mount directory
+			err := os.MkdirAll(brickMountDir, os.ModeDir|os.ModePerm)
+			if err != nil {
+				c.Logger().WithError(err).
+					WithField("path", brickMountDir).
+					Error("Failed to create brick mount directory")
+				return err
+			}
+
+			// Create Logical Volume
+			cmd := exec.Command("lvcreate",
```

Comment:
Can we move commands outside of the transaction like https://github.com/gluster/glusterd2/pull/603/files#diff-ee50b0428ecdd9b14aede7e987db1b59, it makes it easier to handle commands and make our code more modular.

Reply:
Nice, all lv/vg commands can be moved to this file itself. Can we rename that package as `deviceutils` ?

I will move all device commands to that file once that is merged.